turbomoleio/output/files.py

```diff
@@ -425,10 +426,98 @@ def from_parser(cls, parser):
         return cls(numerical_integration=ni, rotational=rot, vibrational=vib, run=run, tm=tm)
 
 
+class Ricc2Output(BaseData):
+    """
+    Object containing the data of the output of a ricc2 calculation.
+    """
+    def __init__(self, mp2, run, tm):
+        """
+        Args:
+            mp2 (MP2Results): MP2 results.
+            run (RunData): information about calculation running (e.g. timings, ...)
+            tm (TurbomoleData): information about the turbomole used for the calculation.
+        """
+        self.mp2 = mp2
+        self.run = run
+        self.tm = tm
+
+    @classmethod
+    def from_parser(cls, parser):
+        """
+        Generates an instance of Ricc2Output from a parser based on the stdout
+        of a Turbomole executable.
+
+        Args:
+            parser (Parser): the parser to be used to extract the data.
+
+        Returns:
+            Ricc2Output.
+        """
+        mp2 = MP2Results.from_parser(parser)
+        run = RunData.from_parser(parser)
+        tm = TurbomoleData.from_parser(parser)
+
+        return cls(mp2=mp2, run=run, tm=tm)
+
+
+class MP2Output(BaseData):
+    """
+    Object containing the data of the output of an MP2 calculation, i.e. mpgrad, ricc2 (with proper MP2 options),
+    or pnoccsd (with proper MP2 options).
+
+    Note: Parsing of PNO-based MP2 calculations (i.e. performed with the pnoccsd program) is not (yet) supported.
+    """
+    def __init__(self, info, energy, geometry, basis, symmetry, cosmo, run, tm):
+        """
+        Args:
+            info (MP2Data): Information about MP2 calculation.
+            energy (float): MP2 energy.
+            geometry (GeometryData): the geometry of the system.
+            basis (BasisData): the basis used for the calculation.
+            symmetry (SymmetryData): information about the symmetry of the molecule.
+            cosmo (CosmoData): cosmo approximations and results.
+            run (RunData): information about calculation running (e.g. timings, ...)
+            tm (TurbomoleData): information about the turbomole used for the calculation.
+        """
+        self.info = info
+        self.energy = energy
+        self.geometry = geometry
+        self.basis = basis
+        self.symmetry = symmetry
+        self.cosmo = cosmo
+        self.run = run
+        self.tm = tm
+
+    @classmethod
+    def from_parser(cls, parser):
+        """
+        Generates an instance of MP2Output from a parser based on the stdout
+        of a Turbomole executable.
+
+        Args:
+            parser (Parser): the parser to be used to extract the data.
+
+        Returns:
+            MP2Output.
+        """
+        info = MP2Data.from_parser(parser)
+        mp2_data = MP2Results.from_parser(parser)
+        energy = mp2_data.energy if mp2_data else None
```

Comment:
this portion of the code seems a bit confusing `info` is and `MP2Data`, and `mp2_data` is an `MP2Results`. Is this correct?

Reply:
Indeed, I've changed that. I've also removed energy and replaced it with results.